boto3/s3/transfer.py

```diff
@@ -268,6 +268,9 @@ def disable_callback(self):
 
     def seek(self, where):
         self._fileobj.seek(self._start_byte + where)
+        if hasattr(self._callback, '__call__') and self._callback_enabled:
```

Comment:
What's the hasattr check for?  This class already makes the assumption if you provide us a callback we'll assume it's the correct type.